panels/models.py

```diff
@@ -0,0 +1,81 @@
+from panels import *    
+
+
+@Session.model_mixin
+class Attendee:
+    assigned_panelists = relationship('AssignedPanelist', backref='attendee')
+
+
+class Event(MagModel):
+    location    = Column(Choice(c.EVENT_LOCATION_OPTS))
+    start_time  = Column(UTCDateTime)
+    duration    = Column(Integer)   # half-hour increments
+    name        = Column(UnicodeText, nullable=False)
+    description = Column(UnicodeText)
+
+    assigned_panelists = relationship('AssignedPanelist', backref='event')
+
+    @property
+    def half_hours(self):
+        half_hours = set()
+        for i in range(self.duration):
+            half_hours.add(self.start_time + timedelta(minutes=30 * i))
+        return half_hours
+
+    @property
+    def minutes(self):
+        return (self.duration or 0) * 30
+
+    @property
+    def start_slot(self):
+        if self.start_time:
+            return int((self.start_time_local - c.EPOCH).total_seconds() / (60 * 30))
+
+
+class AssignedPanelist(MagModel):
```

Comment:
Excellent question!  As of right now, there's really no compelling reason to do it this way.  My expectation when I original defined this model years ago was that I'd be adding new columns to it.  Kind of like how ``Shift`` is basically a many-to-many relationship between ``Attendee`` and ``Job`` but it has enough other attributes (like whether or not it was worked and what rating was given) that we made it an independent model like this one.

I'm still thinking that we'll end up adding new columns to this model as we make our scheduling code more sophisticated... though offhand I can't think of what that might be, so I might just be needlessly planning ahead for something that will never happen.